app/controllers/api/account.php

```diff
@@ -2353,13 +2353,22 @@
                 // Use current session for events.
                 $queueForEvents->setPayload($response->output($session, Response::MODEL_SESSION));
             }
+
+            $queueForEvents
+                ->setEvent('users.[userId].sessions.[sessionId].delete')
+                ->setParam('userId', $user->getId())
+                ->setParam('sessionId', $session->getId())
+                ->setPayload($response->output($session, Response::MODEL_SESSION))
+                ->trigger();
         }
 
         $dbForProject->deleteCachedDocument('users', $user->getId());
 
         $queueForEvents
+            ->reset()
+            ->setEvent('users.[userId].sessions.delete')
             ->setParam('userId', $user->getId())
-            ->setParam('sessionId', $session->getId());
+            ->setPayload($response->output($session, Response::MODEL_SESSION));
```

Comment:
This will just be the last session from the loop, based on [this comment](https://github.com/appwrite/appwrite/pull/7117/files#diff-93771e9bb8e57cbe7d0c0e91e7aea5d85238a362f51caf957049f1d2ccc4d312R2353), we should use the `current` session here (which is still kinda weird it should probably be the user model here, but at least we keep the behavior the same as now)

Reply:
Changed